ydkgen/printer/go/class_printer.py

```diff
@@ -75,109 +58,79 @@ def _get_class_members(self, clazz, leafs, children):
-
-    # GetBundleYangModelsLocation
-    def _print_yang_models_function(self, clazz):
-        bundle_name = snake_case(self.bundle_name)
-        fp = FunctionPrinter(self.ctx, clazz)
-        rstmt = '%s.GetModelsPath()' % bundle_name
-        fp.quick_print('GetBundleYangModelsLocation', return_type='string', return_stmt=rstmt)
-
-    # GetCapabilitiesTable
-    def _print_get_capabilities_table(self, clazz):
+        data_alias = '%s.EntityData' % fp.class_alias
         bundle_name = snake_case(self.bundle_name)
-        fp = FunctionPrinter(self.ctx, clazz)
-        fp.print_function_header_helper(
-            'GetCapabilitiesTable', return_type='map[string]string')
-        fp.ctx.write("return %s.GetCapabilities() " % (bundle_name))
-        fp.print_function_trailer()
 
-    # GetNamespaceTable
-    def _print_get_namespace_table(self, clazz):
-        bundle_name = snake_case(self.bundle_name)
-        fp = FunctionPrinter(self.ctx, clazz)
-        fp.print_function_header_helper(
-            'GetNamespaceTable', return_type='map[string]string')
-        fp.ctx.write("return %s.GetNamespaces() " % (bundle_name))
+        fp.print_function_header_helper('GetEntityData', return_type='*types.CommonEntityData')
+        fp.ctx.writeln('%s.YFilter = %s.YFilter' % (data_alias, fp.class_alias))
+        fp.ctx.writeln('%s.YangName = "%s"' % (data_alias, fp.clazz.stmt.arg))
+        fp.ctx.writeln('%s.BundleName = "%s"' % (data_alias, self.bundle_name.lower()))
+        fp.ctx.writeln('%s.ParentYangName = "%s"' % (data_alias, clazz.owner.stmt.arg))
+        self._print_segment_path(fp, data_alias)
+        fp.ctx.writeln('%s.CapabilitiesTable = %s.GetCapabilities()' % (data_alias, bundle_name))
+        fp.ctx.writeln('%s.NamespaceTable = %s.GetNamespaces()' % (data_alias, bundle_name))
+        fp.ctx.writeln('%s.BundleYangModelsLocation = %s.GetModelsPath()' % (data_alias, bundle_name))
+        fp.ctx.bline()
+        self._print_children(fp, children, data_alias)
+        self._print_leafs(fp, leafs, data_alias)
+
+        fp.ctx.writeln('return &(%s)' % data_alias)
         fp.print_function_trailer()
 
-    # SetParent
-    def _print_set_parent_function(self, clazz):
-        fp = FunctionPrinter(self.ctx, clazz)
-        stmt = '%s.parent = parent' % fp.class_alias
-        fp.quick_print('SetParent', args='parent types.Entity', stmt=stmt)
-
-    # GetParent
-    def _print_get_parent_function(self, clazz):
-        fp = FunctionPrinter(self.ctx, clazz)
-        rstmt = '%s.parent' % fp.class_alias
-        fp.quick_print('GetParent', return_type='types.Entity', return_stmt=rstmt)
-
-    # GetParentYangName
-    def _print_get_parent_yang_name_function(self, clazz):
-        fp = FunctionPrinter(self.ctx, clazz)
-        rstmt = '"%s"' % clazz.owner.stmt.arg
-        fp.quick_print('GetParentYangName', return_type='string', return_stmt=rstmt)
+    @staticmethod
```

Comment:
Just curious why this should be a staticmethod?

Reply:
This was a codacy suggested style change

Reply:
Thanks, makes sense